runtime/near-vm-errors/src/lib.rs

```diff
@@ -432,27 +432,59 @@ impl std::fmt::Display for HostError {
     }
 }
 
-pub mod hex_format {
-    use hex::{decode, encode};
+/// Type-erased error used to shuttle some concrete error coming from `External`
+/// through vm-logic.
+///
+/// The caller is supposed to downcast this to a concrete error type they should
+/// know. This would be just `Box<dyn Any + Eq>` if the latter actually worked.
+pub struct AnyError {
+    any: Box<dyn AnyEq>,
+}
 
-    use serde::de;
-    use serde::{Deserialize, Deserializer, Serializer};
+impl AnyError {
+    pub fn new<E: Any + Eq + Send + Sync + 'static>(err: E) -> AnyError {
+        AnyError { any: Box::new(err) }
+    }
+    pub fn downcast<E: Any + Eq + Send + Sync + 'static>(self) -> Result<E, ()> {
+        match self.any.into_any().downcast::<E>() {
+            Ok(it) => Ok(*it),
+            Err(_) => Err(()),
+        }
+    }
+}
 
-    pub fn serialize<S, T>(data: T, serializer: S) -> Result<S::Ok, S::Error>
-    where
-        S: Serializer,
-        T: AsRef<[u8]>,
-    {
-        serializer.serialize_str(&encode(data))
+impl PartialEq for AnyError {
+    fn eq(&self, other: &Self) -> bool {
+        self.any.any_eq(&*other.any)
     }
+}
 
-    pub fn deserialize<'de, D, T>(deserializer: D) -> Result<T, D::Error>
-    where
-        D: Deserializer<'de>,
-        T: From<Vec<u8>>,
-    {
-        let s = String::deserialize(deserializer)?;
-        decode(&s).map_err(|err| de::Error::custom(err.to_string())).map(Into::into)
+impl Eq for AnyError {}
+
+impl fmt::Debug for AnyError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        fmt::Debug::fmt(self.any.as_any(), f)
+    }
+}
+
+trait AnyEq: Any + Send + Sync {
```

Comment:
Does this truly need to be using `Any`? Could `std::error::Error` be utilized? It also supports downcasting and is generally probably more accurately conveys the intent of what's happening here.

Reply:
Good question! 

On the one hand, yes, this could use `std::error::Error`, and that does seem like a better fit (it's an error!). On the other hand, the intention here is subtly different -- we are not going to use this as a dynamic error, the sole purpose here is to downcast this to a concrete repr down the line. In this sense, `Any` is a better, because it does strictly what we need, and nothing more. 

Perhaps more important thing here is practicality -- if I use `Error` here, than I'd have to write `Display` for `ExternalError`, but that would effectively be dead code. 

So, I'd say let's keep this as Any. But thanks for the sugestions -- using Error didn't occure to me!